torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -76,13 +76,13 @@ static std::vector<int64_t> seq_to_aten_shape(PyObject *py_seq) {
 PyObject* tensor_to_numpy(const at::Tensor& tensor) {
   if (tensor.device().type() != DeviceType::CPU) {
     throw TypeError(
-      "can't convert non-cpu tensor to numpy. Use Tensor.cpu() to "
-      "copy the tensor to host memory first.");
+      "can't convert %s device type tensor to numpy. Use Tensor.cpu() to "
+      "copy the tensor to host memory first.", tensor.device().str().c_str());
   }
   if (tensor.layout() != Layout::Strided) {
       throw TypeError(
-        "can't convert non-strided tensor to numpy."
-        "convert the tensor to a strided layout first.");
+        "can't convert %s layout tensor to numpy."
+        "convert the tensor to a strided layout first.", c10::str(tensor.layout()).c_str());
```

Comment:
This was a format error on the commit I was working on but rebasing shows this was already fixed in #33612. This does give a more informative error though.

cc @mruberry

Reply:
SGTM.